includes/admin/reports/class-wc-report-downloads.php

```diff
@@ -0,0 +1,251 @@
+				}
+
+				// Product information
+				$product_description = '';
+				if ( $sku = $product->get_sku() ) {
+					$product_description .= $sku . ' - ';
+				}
+				$product_description .= $product->get_name();
+				echo edit_post_link( $product_description, '', '', $product->get_id() );
+
+				// Order information
+				echo ' in Order ';
+				echo edit_post_link( '#' . $permission->order_id, '', '', $permission->order_id );
+
+				echo '</h2>';
+			}
+		}
+
+		echo '<div id="poststuff" class="woocommerce-reports-wide">';
+		$this->display();
+		echo '</div>';
+	}
+
+	/**
+	 * Get column value.
+	 *
+	 * @param mixed $item
+	 * @param string $column_name
+	 */
+	public function column_default( $item, $column_name ) {
+
+		$permission = null;
+		$product = null;
+		try {
+			$permission	= new WC_Customer_Download( $item->permission_id );
+			$product	= wc_get_product( $permission->product_id );
+		} catch ( Exception $e ) {
+			// Ok to continue rendering other information even if permission and/or product is not found
+		}
+
+		switch ( $column_name ) {
+			case 'timestamp' :
+				echo esc_html( $item->timestamp );
+				break;
+
+			case 'product' :
+				if ( empty( $product ) ) {
+					break;
+				}
+
+				$product_description = '';
+				if ( $sku = $product->get_sku() ) {
+					$product_description .= $sku . ' - ';
+				}
+				$product_description .= $product->get_name();
+
+				echo edit_post_link( $product_description, '', '', $product->get_id() );
+
+				break;
+
+			case 'order' :
+				if ( ! empty( $permission ) ) {
+					echo edit_post_link( '#' . $permission->order_id, '', '', $permission->order_id );
+				}
+			
+				break;
+
+			case 'user' :
+				if ( $item->user_id > 0 ) {
+					$user = get_user_by( 'id', $item->user_id );
+
+					if ( ! empty( $user ) ) {
+						$user_description = $user->data->user_nicename;
+						if ( empty( $user_description ) ) {
+							$user_description = $user->data->user_email;
+						}
+
+						echo '<a href="' . get_edit_user_link( $item->user_id ) . '">' . esc_html( $user_description ) . '</a>';
+					}
+				}
+
+				break;
+
+			case 'user_ip_address' :
+				echo esc_html( $item->user_ip_address );
+
+				break;
+		}
+	}
+
+	/**
+	 * Get columns.
+	 *
+	 * @return array
+	 */
+	public function get_columns() {
+
+		$columns = array(
+			'timestamp'			=> __( 'Download time', 'woocommerce' ),
+			'product'			=> __( 'Product', 'woocommerce' ),
+			'order'				=> __( 'Order', 'woocommerce' ),
+			'user'				=> __( 'User', 'woocommerce' ),
+			'user_ip_address'	=> __( 'User IP Address', 'woocommerce' ),
+		);
+
+		return $columns;
+	}
+
+	/**
+	 * Prepare download list items.
+	 */
+	public function prepare_items() {
+
+		$this->_column_headers = array( $this->get_columns(), array(), $this->get_sortable_columns() );
+		$current_page          = absint( $this->get_pagenum() );
+		$per_page              = apply_filters( 'woocommerce_admin_downloads_report_downloads_per_page', 20 );
```

Comment:
Since this is filterable, let's ensure user cannot set `0` because it's used in division below.